### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by mhallsmoore**
This is a reasonably large update to remove old examples and update others to use the current API.

I've modified how the Backtest object works, but only in so much as to utilise default PositionSizer, Compliance, RiskManager etc objects so that the ***_backtest.py files are not cluttered with many class initialisations.

Be warned - this does change the Backtest API slightly! Perhaps the best way to test it is to head to the examples directory and look at "moving_average_cross_backtest.py" or "buy_and_hold_backtest.py", which now work with the updated API.

I also fixed an insidious bug that was causing backtest results to be non-deterministic. This happened because Pandas was stochastically sorting the Yahoo prices DataFrame. These prices were then being filtered into the queue in random order across runs.

TearsheetStatistics now also functions when no trades have been placed. Previously, an exception was generated if this was the case. In particular, the Buy-and-Hold example will now function under a tearsheet example.

### CODE REVIEW

File: examples/buy_and_hold_backtest.py [hunk 1]
```diff
@@ -1,81 +1,70 @@
...
+def run(config, testing, tickers, filename):
+    # Backtest information
+    title = ['Buy and Hold Example on %s' % tickers[0]]
+    initial_equity = 10000.0
+    start_date = datetime.datetime(2000, 1, 1)
+    end_date = datetime.datetime(2014, 1, 1)
 
-    # Use the default Statistics
-    statistics = SimpleStatistics(config, portfolio_handler)
+    # Use the MAC Strategy
```
Comment (ryankennedyio) [inline examples/buy_and_hold_backtest.py]: Comment needs an update

File: qstrader/trading_session/backtest.py [hunk 2]
```diff
@@ -10,29 +18,81 @@ class Backtest(object):
...
-        statistics, equity,
-        sentiment_handler=None
+        self, config, strategy, tickers,
+        equity, start_date, end_date, events_queue,
+        price_handler=None, portfolio_handler=None,
+        compliance=None, position_sizer=None,
+        execution_handler=None, risk_manager=None,
+        statistics=None, sentiment_handler=None,
+        title=None, benchmark=None
     ):
```
Comment (ryankennedyio) [inline qstrader/trading_session/backtest.py]: Way nicer.

File: qstrader/trading_session/backtest.py [hunk 3]
```diff
@@ -10,29 +18,81 @@ class Backtest(object):
...
         """
         Set up the backtest variables according to
         what has been passed in.
         """
-        self.price_handler = price_handler
+        self.config = config
         self.strategy = strategy
```
Comment (ryankennedyio) [inline qstrader/trading_session/backtest.py]: Has this been tested with combined (i.e. multiple at once) strategies?

Comment (mhallsmoore) [inline qstrader/trading_session/backtest.py]: I presume you mean multiple derived Strategy class objects for the same Backtest? If so, I don't think so. If you have an idea of the API that you'd like to see for that, I could include it in this branch and then include an example set of strategies to test it.

Let me know ;-)

File: examples/buy_and_hold_backtest.py (hunk 1 above)
Comment (mhallsmoore) [inline examples/buy_and_hold_backtest.py]: Thanks! Have updated locally.

**Comment by mhallsmoore** [comment]
I've just added some of the older Position fixes into this branch. There was an error in how the Realised PnL was being calculated. In addition, the Portfolio itself was miscalculating its equity value, based on the incorrect handling of the Realised PnL. 

They have both been fixed in this update. @femtotrader's tests for the Position were also added in and now pass.

**Pull request merged**